### DISCUSSION TITLE
Help with compiling and running?

### DISCUSSION THREAD

**Discussion by Anthdawg5443** [Kiiyya/BattleFox#39] [General]
i cannot get this to run maybe im doing something wrong but this is the error i get while i try to run the program :/ sorry to be a bother C:\Users\Anthony\Desktop\BattleFox\target\debug>battlefox.exe
22:17:39 INFO  battlefox\src\main.rs:237 > This is BattleFox map-exclusion-test-14-g45b1fda
Log is written to logs\battlefox_2022-12-07_22-17-39.log
22:17:39 INFO  battlefox\src\main.rs:96 > Loading configs/admins.yaml
Error: Failed to open config file

Caused by:
    The system cannot find the path specified. (os error 3)

**Comment by Razer2015** [comment]
Hi, you need to have [this](https://github.com/Kiiyya/BattleFox/tree/main/battlefox/configs) `configs` folder and its content in the same directory as the battlefox.exe to run it.

I've been running it in Docker or developing with `cargo run`. But I think you also need to set some environment variables to run it properly. I believe at least the `BFOX_RCON_*` and `DATABASE_URL` are required. Database stuff in the current version is a bit broken because the migration from diesel to sqlx is still WIP, but ban_enforcer uses the AdKats tables. So if you want to use the ban_enforcer, your database needs to have the AdKats schema.
```
BFOX_RCON_IP=<yourRCONIP>
BFOX_RCON_PORT=47200
BFOX_RCON_PASSWORD=<yourRCONPassword>
DATABASE_URL=mysql://username:password@127.0.0.1:3306/bfox_test_db
RUST_LOG="warn,battlefox=trace,battlefield_rcon=debug"
TZ=Europe/Helsinki
```

Let me know if  you get it working or if there's other issues after these.

**Reply by Anthdawg5443** [reply]
got it working its just as cool as i thought it would be im looking forward to more progress let me know if you guys have any donations or anything setup to help with the cause :)

**Comment by Kiiyya** [comment]
By the way, you can just do `cargo run` instead of running the executable from the `target/` folder yourself. `cargo` will then build the project if necessary.

**Reply by Anthdawg5443** [reply]
Yeah I'm doing cargo run --bin battlefox and it compiles and runs great just was figuring out the commands and the structure of it all but its looking great and less of a hassle than procon and if it includes all that's showing it'll be a pretty efficient replacement for it that won't crash/hang very often require restarts that could last minutes ext have persistent votemap maybe so if the layer restarts it doesn't mess up a votemap that took place before the layer restart stuff like that but so far it's pretty impressive and I'm glad you guys are as active as you're I ran a server for a year and a half had around 90k players from 2019 to 2020
